.ci/setup.sh

```diff
@@ -71,9 +71,40 @@ else  # Linux
         sudo apt-get install --no-install-recommends -y \
             libboost1.74-dev \
             ocl-icd-opencl-dev
+    fi
+    if [[ $TASK == "gpu" || $TASK == "bdist" ]]; then
         if [[ $IN_UBUNTU_LATEST_CONTAINER == "true" ]]; then
+            sudo apt-get update
             sudo apt-get install --no-install-recommends -y \
                 pocl-opencl-icd
+        elif [[ $(uname -m) == "aarch64" ]]; then
+            yum install -y \
+                epel-release \
+                gcc-c++ \
+                hwloc-devel \
+                sudo
+            yum install -y \
+                llvm-toolset-7.0-clang-devel \
+                llvm-toolset-7.0-llvm-devel \
+                ocl-icd-devel
+            git clone --depth 1 --branch v1.8 https://github.com/pocl/pocl.git
```

Comment:
In the time since this PR was first opened, PoCL has put up a new major release (v3.0, July 2022): https://github.com/pocl/pocl/releases.

Could you try upgrading to that, to take advantage of bugfixes and improvements in that project? I think that right now, while this support is new and we're actively testing it, is a great time to try to use the newest version possible.

Or, if that's a large amount of effort, after merging this would you please write up a new issue documenting the work to upgrade to that new version of PoCL?

Reply:
Thanks for letting me know, I'll take a look at upgrading to PoCL v3.0!

For this specific PR, I have only aligned linux aarch64 to also use PoCL like we do on linux x86-64. I would thus suggest leaving the upgrade out of the scope of this PR.

Reply:
Ok nice, makes sense! Once we merge this PR, would you please document a feature request like "upgrade to PoCL v3.0" in this project's issue tracker? Just so we don't forget about. it.

Reply:
I created #5596 so we don't forget about this